code/game/turfs/simulated/floor/light_floor.dm

```diff
@@ -24,9 +31,21 @@
 		set_light(0)
 
 /turf/simulated/floor/light/BeforeChange()
-	set_light(0)
+	toggle_light(FALSE)
 	..()
 
+/turf/simulated/floor/light/proc/power_update()
+	SIGNAL_HANDLER
+	if(power_check() || !on)
+		return
+	extinguish_light()
```

Comment:
use `toggle_light`  extinguish light will spam messages if there is a bunch of tiles